app/models/mixins/miq_provision_quota_mixin.rb

```diff
@@ -375,7 +375,13 @@ def storage(prov, cloud, vendor, flavor_obj = nil)
       return nil unless flavor_obj
       flavor_obj.root_disk_size.to_i + flavor_obj.ephemeral_disk_size.to_i + flavor_obj.swap_disk_size.to_i
     else
-      prov.kind_of?(MiqRequest) ? prov.vm_template.provisioned_storage : prov.miq_request.vm_template.provisioned_storage
+      infra_storage(prov.kind_of?(MiqRequest) ? prov : prov.miq_request).to_i
+    end
+  end
+
+  def infra_storage(request)
```

Comment:
also, could you use verb names for methods that do stuff rather than nouns.
(sorry a nit but makes a difference for future developers)